cmd/packer-sdc/internal/fix/fix.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fix
+
+// Fixer applies all defined fixes on a plugin dir; a Fixer should be idempotent.
+type fixer interface {
+	Fix(filename string, data []byte) ([]byte, error)
+}
+
+type fix struct {
+	name, description string
+	fileGlob          string
```

Comment:
I wonder, do we need this one? It feels a bit limiting for future fixes maybe since globs may not recursively look for files to glob, and right now it's completely static, so I wonder if we need to define this as part of `fix` right now

Reply:
So for each of the fixers I'm looking to in to help identify what file(s) to apply the fix on. Do you have a suggest on how I should handle? Or are you suggesting just statically pointing to the file since it is just one file in the project at this time?

Reply:
I understand the objective for this attribute, but I think that for now, this is maybe too early to introduce this glob as the way to get the file(s) to work on, and I am worried it may be limited in the end.
Maybe this could be a function instead if we want to be able to add a new fixer without having to specify something similar later? I think a function would be more futureproof for this.

Reply:
Ah I like this so maybe a function that takes a directory and returns the files to apply the fix on. Is that what you are thinking?

Reply:
Essentially yes!

Reply:
Done tell me what you think?

Reply:
Looks good to me!